crates/astria-sequencer/src/ibc/ics20_transfer.rs

```diff
@@ -581,20 +580,27 @@ async fn execute_ics20_transfer_bridge_lock<S: StateWriteExt>(
     }
 
     // assert memo is valid
-    let deposit_memo: Ics20TransferDepositMemo =
+    let deposit_memo: memos::v1alpha1::Ics20TransferDeposit =
         serde_json::from_str(&memo).context("failed to parse memo as Ics20TransferDepositMemo")?;
 
     ensure!(
-        !deposit_memo.rollup_address.is_empty(),
+        !deposit_memo.rollup_deposit_address.is_empty(),
         "packet memo field must be set for bridge account recipient",
     );
 
     ensure!(
-        deposit_memo.rollup_address.len() <= MAX_ROLLUP_ADDRESS_BYTE_LENGTH,
+        deposit_memo.rollup_deposit_address.len() <= MAX_ROLLUP_ADDRESS_BYTE_LENGTH,
         "rollup address is too long: exceeds MAX_ROLLUP_ADDRESS_BYTE_LENGTH",
     );
```

Comment:
should this validation now be handled by the core library?

Reply:
I wanted that before but got pushback for it (here and for the one above).

IMO it's fine to just leave it as a stupid container type.